test/test_quantization.py

```diff
@@ -0,0 +1,276 @@
+from __future__ import absolute_import, division, print_function, unicode_literals
+import torch
+import torch.nn.quantized as nnq
+import torch.quantization as tq
+from torch.quantization import \
+    observer, Observer, WeightObserver, QConfig, QuantWrapper
+from common_utils import TestCase, run_tests
+
+class SingleLinearModel(torch.nn.Module):
+    def __init__(self):
+        super(SingleLinearModel, self).__init__()
+        self.fc1 = torch.nn.Linear(5, 5).to(dtype=torch.float)
+
+    def forward(self, x):
+        x = self.fc1(x)
+        return x
+
+class Model2(torch.nn.Module):
+    def __init__(self):
+        super(Model2, self).__init__()
+        self.fc1 = torch.nn.Linear(5, 5).to(dtype=torch.float)
+        self.fc2 = torch.nn.Linear(5, 5).to(dtype=torch.float)
+
+    def forward(self, x):
+        x = self.fc1(x)
+        x = self.fc2(x)
+        return x
+
+class Model3(torch.nn.Module):
+    def __init__(self):
+        super(Model3, self).__init__()
+        self.fc = torch.nn.Linear(5, 5).to(dtype=torch.float)
+
+    def forward(self, x):
+        x = self.fc(x)
+        return x
+
+class NestedModel(torch.nn.Module):
+    def __init__(self):
+        super(NestedModel, self).__init__()
+        self.sub1 = Model3()
+        self.sub2 = Model2()
+        self.fc3 = torch.nn.Linear(5, 5).to(dtype=torch.float)
+
+    def forward(self, x):
+        x = self.sub1(x)
+        x = self.sub2(x)
+        x = self.fc3(x)
+        return x
+
+def eval_fn(model, data):
+    model(data)
+
+default_options = {
+    'dtype': torch.qint8,
+    'qscheme': torch.per_tensor_affine
+}
+
+custome_options1 = {
+    'dtype': torch.quint8,
+    'qscheme': torch.per_tensor_affine
+}
+
+class ModelQuantizeAPITest(TestCase):
+    def test_quantize1(self):
+        '''
+            Quantize SingleLinearModel which has one Linear module, make sure it is swapped
+            to nnq.Linear which is the quantized version of the module
+        '''
+        calib_data = torch.rand(20, 5, dtype=torch.float)
+        model = SingleLinearModel()
+        qConfigDict = {
+            # Top level qconfig
+            '': QConfig(weight=observer(WeightObserver, default_options),
```

Comment:
should there be a default QConfig thingy that just returns a pair weight/activations with default arguments? (and maybe allow to override dtype / qscheme for them). It's a bit verbose to spell it out like this